### ISSUE TITLE
Add Derecho as a supported platform

### ISSUE THREAD

**Issue by mark-a-potts** [ufs-community/ufs-weather-model#1835]
## Description
NCAR is getting ready to replace Cheyenne with Derecho. The UFS-WM needs to be updated to support the new system.

## Solution
Spack-stack v. 1.4.1 has been installed on Derecho with Intel 2021.1.8 compilers and both intel-mpi and cray-mpich. Derecho shares a filesystem with Cheyenne, so data for regression tests is already accessible. Module files and associated regression testing scripts need to be updated to include the new platform.

## Related to
Example:
- The following PR needs to be merged before this can be addressed ufs-community/ufs-weather-model/pull/<1707>

**Comment by mkavulich** [comment]
@mark-a-potts Is there a timeline on this? I am curious about what barriers still exist to transferring work over to Derecho since the model already compiles and runs successfully on that machine.

**Comment by mark-a-potts** [comment]
I think that SRW should be able to support derecho now as well, if that is what you are asking.

**Comment by zach1221** [comment]
ufs-wm PR #1836 is merged into ufs-community:develop.
